src/gt4py/next/otf/stages.py

```diff
@@ -32,11 +33,24 @@
 SettingT_co = TypeVar("SettingT_co", bound=languages.LanguageSettings, covariant=True)
 
 
+@dataclasses.dataclass(frozen=True)
+class ProgramDefinition:
+    definition: types.FunctionType
+    grid_type: Optional[common.GridType] = None
+
+
+@dataclasses.dataclass(frozen=True)
+class ProgramPast:
```

Comment:
I don't know of a really good name, but maybe `PastProgramDefinition`?

Reply:
👍